src/expr.rs

```diff
@@ -1892,6 +1904,7 @@ fn choose_rhs<R: Rewrite>(
     shape: Shape,
     orig_rhs: Option<String>,
     rhs_tactics: RhsTactics,
+    has_rhs_comment: bool,
```

Comment:
Perhaps there is a better way than adding a flag. But I can't think of a good idea ...

Reply:
What about inspecting the content of `orig_rhs` and see if it starts with the comment? Something like the following:

```rust
let has_rhs_comment = orig_rhs.map_or(false, |s| starts_with_comment(s.trim_left()));
```

Reply:
I tried your proposal, but it's failed. The orig_rhs doesn't have the comments. The lhs has the comment and indent insteadof orig_rhs. below is an example.

```
fn main() {
    let after =
    // after_comment
    if true {
        1.0
    };
}
// &lhs = "let after =\n    // after_comment\n    "
// &orig_rhs = Some("if true {\n        1.0\n    }",)
```

lhs is [this](https://github.com/rust-lang/rustfmt/pull/3857/files#diff-180bb3ad456cc10517d878fb7795341cR1862)

Reply:
I see, thank you for your explanation. In that case, is it possible to use the `lhs` to detect comments instead? Something like the following:

```rust
let has_rhs_comment = !lhs.trim_right().ends_with_comment();
```

Reply:
Absolutely. It's a good idea. I fixed it.